ext/ddtrace_profiling_native_extension/extconf.rb

```diff
@@ -124,6 +126,34 @@ def add_compiler_flag(flag)
 EXTENSION_NAME = "ddtrace_profiling_native_extension.#{RUBY_VERSION}_#{RUBY_PLATFORM}".freeze
 
 if CAN_USE_MJIT_HEADER
+  mjit_header_file_name = "rb_mjit_min_header-#{RUBY_VERSION}.h"
+
+  # Validate that the mjit header can actually be compiled on this system. We learned via
+  # https://github.com/DataDog/dd-trace-rb/issues/1799 and https://github.com/DataDog/dd-trace-rb/issues/1792
+  # that even if the header seems to exist, it may not even compile.
+  # `have_macro` actually tries to compile a file that mentions the given macro, so if this passes, we should be good to
+  # use the MJIT header.
+  # Finally, the `COMMON_HEADERS` conflict with the MJIT header so we need to temporarily disable them for this check.
+  original_common_headers = MakeMakefile::COMMON_HEADERS
+  MakeMakefile::COMMON_HEADERS = ''.freeze
+  unless have_macro('RUBY_MJIT_H', mjit_header_file_name)
```

Comment:
Is this the line where we compile the native mjit headers?
I was looking for some sort of "compile" code line, but couldn't find which one it was.

Reply:
Yup, the magic is all implemented by Ruby's mkmf:

1. https://github.com/ruby/ruby/blob/7fc9d83bd1c1b8c44790b6af0f81f3b7364270ca/lib/mkmf.rb#L1040
2. https://github.com/ruby/ruby/blob/7fc9d83bd1c1b8c44790b6af0f81f3b7364270ca/lib/mkmf.rb#L898

...and if you follow it down the rabbit hole it eventually shells out to the compiler and checks if it was able to perform the compile.